src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,1008 +50,333 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
-    * @description utility to copy Address fields from a Contact or Account to an Address object.
-    * Handles instances where State and Country picklists are enabled, and multiline street addresses.
-    * @param sobjSrc the source Contact or Account
-    * @param strFieldPrefixSrc the address fields to copy from, ie., Mailing, Other, Shipping, Billing
-    * @param addr the Address object to copy to
-    * @param strAddrType if not null, copy to address.Address_Type__c
-    * @return void
-    */ 
-    public static void copyAddressStdSObjAddr(sobject sobjSrc, string strFieldPrefixSrc, Address__c addr, string strAddrType) {
-        addr.MailingStreet__c = (string)sobjSrc.get(strFieldPrefixSrc + 'Street');
-        handleMultilineStreet(addr);
-        addr.MailingCity__c = (string)sobjSrc.get(strFieldPrefixSrc + 'City');
-        addr.MailingPostalCode__c = (string)sobjSrc.get(strFieldPrefixSrc + 'PostalCode');
-        addr.MailingState__c = (string)sobjSrc.get(strFieldPrefixSrc + 'State');
-        addr.MailingCountry__c = (string)sobjSrc.get(strFieldPrefixSrc + 'Country');
-        addr.Geolocation__Latitude__s = (decimal)sobjSrc.get(strFieldPrefixSrc + 'Latitude');
-        addr.Geolocation__Longitude__s = (decimal)sobjSrc.get(strFieldPrefixSrc + 'Longitude');
-        
-        // try to find state and country by code if they aren't specified?
-        if (isStateCountryPicklistsEnabled) {
-
-            // lookup country if needed, by country code
-            string strCountryCode = (string)sobjSrc.get(strFieldPrefixSrc + 'CountryCode');
-            if (addr.MailingCountry__c == null && strCountryCode != null) {
-                addr.MailingCountry__c = validCountriesByCode.get(strCountryCode);
-            }
-            
-            // lookup state if needed, by country code and state code.
-            string strStateCode = (string)sobjSrc.get(strFieldPrefixSrc + 'StateCode');
-            if (addr.MailingState__c == null && strCountryCode != null && strStateCode != null) {
-                // assume it's a valid statecode
-                addr.MailingState__c = strStateCode;
-            }
-        }
-        
-        // if specified, also include Address Type
-        if (strAddrType != null) 
-            addr.Address_Type__c = strAddrType;
-    }
-
-    /*******************************************************************************************************
-    * @description Utility to copy Address fields from an Address object to a Contact or Account.
-    * Handles instances where State and Country picklists are enabled, and multiline street addresses.
-    * @param addr the Address object to copy from
-    * @param sobjDst the destination Contact or Account
-    * @param strFieldPrefix the address fields to copy to, ie., Mailing, Other, Shipping, Billing
-    * @param strFieldAddrType an optional Address Type field on sobjDst to copy to 
-    * @return void
```

Comment:
Nesting too many `IF` statements affects readability and makes maintenance more difficult. I suggest reviewing the design to simplify the flow of this code block.